### PR TITLE
Remove nucleus

### PR DESCRIPTION
**Pull request by MarkVaughn**
To avoid dependency hell I removed nucleus from this repo, this is the start of a 2.0 branch (where we will integrate new slapp)

### CODE REVIEW

File: src/SellerLabs/Slapp/Entities/BaseEntity.php [hunk 1]
```diff
@@ -39,4 +39,140 @@ protected function convertDatetime($value)
 
         return Carbon::createFromFormat(static::DATETIME_FORMAT, $value);
     }
+
```
Comment (MarkVaughn) [inline src/SellerLabs/Slapp/Entities/BaseEntity.php]: All these methods come from the parent class that I removed.

File: tests/SellerLabs/Slapp/HelperTest.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+     * @dataProvider snakeProvider
+     *
+     * @param string $value
+     * @param string $expect
+     */
+    public function testSnakeCase($value, $expect)
+    {
+        $this->assertEquals($expect, snake_case($value));
+    }
+}
```
Comment (dsh) [inline tests/SellerLabs/Slapp/HelperTest.php]: What is the purpose of these helpers? Are these used only by SLAPP, or filling in where you removed Nucleus stuff?

File: src/SellerLabs/Slapp/Exceptions/SlappClientException.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * This file is part of the Snagshout package
+ */
+
+namespace SellerLabs\Slapp\Exceptions;
+
+use Exception;
+
+/**
+ * Class SlappClientException.
```
Comment (MarkVaughn) [inline src/SellerLabs/Slapp/Exceptions/SlappClientException.php]: Replaces CoreException/LackOfCoffeeException

File: src/SellerLabs/Slapp/helpers.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * Copyright 2014-2017, SellerLabs <snagshout-devs@sellerlabs.com>
```
Comment (MarkVaughn) [inline src/SellerLabs/Slapp/helpers.php]: Replaces all the Str/Arr functions of nucleus

File: tests/SellerLabs/Slapp/HelperTest.php (hunk 2 above)
Comment (MarkVaughn) [inline tests/SellerLabs/Slapp/HelperTest.php]: Filling in where nucleus things were removed. They are based on the laravel helper functions of the same name.

**Pull request merged**